central/auth/service/service_impl_test.go

```diff
@@ -126,3 +181,428 @@ func (s *authServiceAccessControlTestSuite) TestAuthServiceResponse() {
+					Role:  testRole3,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	config2, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: &v1.AuthMachineToMachineConfig{
+			TokenExpirationDuration: "1h",
+			Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+			Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+				{
+					Key:   "sub",
+					Value: "somewhere",
+					Role:  testRole1,
+				},
+				{
+					Key:   "aud",
+					Value: "the",
+					Role:  testRole2,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	configs, err := s.svc.ListAuthMachineToMachineConfigs(s.accessCtx, nil)
+	s.NoError(err)
+
+	s.ElementsMatch(configs.GetConfigs(), []*v1.AuthMachineToMachineConfig{config1.GetConfig(), config2.GetConfig()})
+}
+
+func (s *authServiceAccessControlTestSuite) TestUpdateConfig() {
+	config, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: &v1.AuthMachineToMachineConfig{
+			TokenExpirationDuration: "1h",
+			Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+			Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+				{
+					Key:   "sub",
+					Value: "something",
+					Role:  testRole1,
+				},
+				{
+					Key:   "aud",
+					Value: "github",
+					Role:  testRole3,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	config.GetConfig().Mappings = []*v1.AuthMachineToMachineConfig_Mapping{
+		{
+			Key:   "sub",
+			Value: "someone",
+			Role:  testRole2,
+		},
+	}
+
+	updatedConfig, err := s.svc.UpdateAuthMachineToMachineConfig(s.accessCtx,
+		&v1.UpdateAuthMachineToMachineConfigRequest{Config: config.GetConfig()})
+	s.NoError(err)
+	s.Equal(config.GetConfig(), updatedConfig.GetConfig())
+
+	newConfig := &v1.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		TokenExpirationDuration: "1m",
+		Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+	}
+	updatedConfig, err = s.svc.UpdateAuthMachineToMachineConfig(s.accessCtx, &v1.UpdateAuthMachineToMachineConfigRequest{
+		Config: newConfig,
+	})
+	s.NoError(err)
+	s.Equal(updatedConfig.GetConfig(), newConfig)
+}
+
+func (s *authServiceAccessControlTestSuite) TestRemoveConfig() {
+	config, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: &v1.AuthMachineToMachineConfig{
+			TokenExpirationDuration: "1h",
+			Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+			Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+				{
+					Key:   "sub",
+					Value: "something",
+					Role:  testRole1,
+				},
+				{
+					Key:   "aud",
+					Value: "github",
+					Role:  testRole3,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	_, err = s.svc.DeleteAuthMachineToMachineConfig(s.accessCtx, &v1.ResourceByID{Id: "80c053c2-24a7-4b97-bd69-85b3a511241e"})
+	s.NoError(err)
```

Comment:
If it's expected that the deletion of a non-existent config succeeds, I suggest to mention that in the `.proto` description of `DeleteAuthMachineToMachineConfig` api.

I also suggest to factor this as a separate test for explicitness. Again, the idea is to make the behavior explicit so that the future maintainers who'd change the code and alter the behavior would think twice if the alteration will break any existing integrations.

Or is it just a side effect of Update test adding a config with this `"80c053c2-24a7-4b97-bd69-85b3a511241e"` id and Delete test running after that?

Reply:
No, it's a behavior of the postgres store.

I'll add a comment on the proto file and moved it to a separate test for explicitness.